library/src/main/java/com/pokegoapi/main/AsyncServerRequest.java

```diff
@@ -15,60 +15,93 @@
 
 package com.pokegoapi.main;
 
-import com.google.protobuf.GeneratedMessage;
-
 import POGOProtos.Networking.Requests.RequestOuterClass.Request;
 import POGOProtos.Networking.Requests.RequestTypeOuterClass.RequestType;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.GeneratedMessage;
+import com.pokegoapi.api.PokemonGo;
+import com.pokegoapi.util.PokeAFunc;
+import com.pokegoapi.util.PokeCallback;
 import lombok.Getter;
 
+import java.util.ArrayList;
+import java.util.Collections;
+
 /**
  * The type Server request.
  */
-public class AsyncServerRequest {
-	@Getter
-	private final long id = System.nanoTime();
+public class AsyncServerRequest<T extends GeneratedMessage, K> {
 	@Getter
 	private final RequestType type;
 	@Getter
 	private final Request request;
 	@Getter
-	private final boolean requireCommonRequest;
+	final ArrayList<InternalServerRequest> boundedRequests = new ArrayList<>();
```

Comment:
Any reason for the left hand ArrayList? Why an array list here, would LinkedList make much more sense?